src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -2232,13 +2232,13 @@ private bool ConsumeComment()
                 }
                 else
                 {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWithinStartOfComment);
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
```

Comment:
Use `InvalidCharacterAtStartOfComment` here. `UnexpectedEndOfDataWhileReadingComment` doesn't make sense for this case.